src/redux/user/actions.js

```diff
@@ -26,6 +26,33 @@ export const loadWritePermissions = (getUserPermissions) => (dispatch) => {
     );
 };
 
+export const loadReadIntegrationsEndpointsPermissions = (getUserPermissions) => (dispatch) => {
+  dispatch({ type: ACTION_TYPES.SET_READ_INTEGRATIONS_ENDPOINTS_PERMISSIONS_PENDING });
+
+  return getUserPermissions('integrations', true) // bypassCache = true
```

Comment:
What you can do is actually flip the order of actions and dispatching a promise.
```JS
dispatch({
  type: ACTION_TYPES.SET_READ_INTEGRATIONS_ENDPOINTS_PERMISSIONS,
  payload: getUserPermissions('integrations', true).then((permissions) => {
    const allPermission = permissions.map((curr) => curr?.permission);
    return allPermission.includes('integrations:*:*') || allPermission.includes('integrations:endpoints:read');
  })
})
```

Reply:
really cool! I didn't know you could do this